src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs

```diff
@@ -0,0 +1,44 @@
+using FluentValidation;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Nhs.Appointments.Api.Auth;
+using Nhs.Appointments.Api.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Nhs.Appointments.Api.Functions
+{
+    public class GetUserProfileFunction(IValidator<EmptyRequest> validator, IUserContextProvider userContextProvider, ILogger<GetSitesForUserFunction> logger)
+    : BaseApiFunction<EmptyRequest, UserProfile>(validator, userContextProvider, logger)
+    {
+
+        [OpenApiOperation(operationId: "GetUserProfile", tags: new[] { "Utility" }, Summary = "Gets information about the signed in user")]
+        [OpenApiRequestBody("text/json", typeof(SetBookingStatusRequest))]
```

Comment:
Looks like this request type might be a copypasta mistake :)

Reply:
Indeed! I have resolved this in the very next PR which includes changes to this endpoint